resources/metadata/detect.yaml

```diff
@@ -172,6 +172,35 @@ spec:
           - major-minor
           - semantic
           - full
+      - name: projectSettingsFile
+        type: string
+        description: "Path or url to the mvn settings file that should be used as project settings file."
+        scope:
+          - PARAMETERS
+          - STAGES
+          - STEPS
+        mandatory: false
+        default:
+      - name: globalSettingsFile
+        type: string
+        description: "Path or url to the mvn settings file that should be used as global settings file"
+        scope:
+          - PARAMETERS
+          - STAGES
+          - STEPS
+        mandatory: false
+        default:
```

Comment:
Please tell me if my guess is correct: The contents of the settings file(s) for detect may differ from the contents of other maven steps. That is the reason why the GENERAL scope is not specified and no aliases. And it's also the reason why you introduced the `overwrite` flag to the `downloadSettingsIf/FromUrl()` functions. If so, then any maven steps running after the detect step will now use the wrong settings. I see several solutions: Always overwrite, specify a unique filename depending on step, append a hash of the URL to the filename (assuming that the same URL will result in the same file contents).

Reply:
Was just wrong. GENERAL was missing.

Reply:
Then the aliases are missing as well.